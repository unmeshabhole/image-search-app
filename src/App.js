import React, { useEffect, useState } from "react";
import { getImages, searchImages } from "./services/imageSearchService";

import Gallery from "./components/gallery/gallery";
import SearchBox from "./components/searchBox/searchBox";

import "./App.css";

const App = () => {
  const [search, setSearch] = useState("");
  const [images, setImages] = useState([]);
  const [loading, setLoading] = useState(false);
  const [error, setError] = useState(null);

  const getImageData = async () => {
    const { data, error } = await getImages();
    handleData(data, error);
  };

  const searchImageData = async (search) => {
    const { data, error } = await searchImages(search);
    handleData(data, error);
  };

  const handleData = (data, error) => {
    setLoading(true);
    if (data) {
      setImages(data);
    } else if (error) {
      setError(error);
    }
    setLoading(false);
  };

  // get images using hooks
  useEffect(async () => {
    if (search.length > 0) {
      searchImageData(search);
    } else {
      getImageData();
    }
  }, [search]);

  return (
    <center>
      <h1>Image Search App</h1>
      <SearchBox search={search} setSearch={setSearch} />
      <div className="image-container">
        {loading ? (
          <p>Loading</p>
        ) : error ? (
          <p>Error loading data</p>
        ) : images.length > 0 ? (
          <Gallery images={images} />
        ) : (
          <p>No images found</p>
        )}
      </div>
    </center>
  );
};

export default App;
