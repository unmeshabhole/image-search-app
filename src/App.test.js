import * as React from "react";
import * as enzyme from "enzyme";

import { mount, shallow } from "enzyme";
import { render, screen } from "@testing-library/react";

import Adapter from "enzyme-adapter-react-16";
import App from "./App";
import Gallery from "./components/gallery";
import SearchBox from "./components/searchBox";

enzyme.configure({ adapter: new Adapter() });

test("should render page header", () => {
  render(<App />);
  const linkElement = screen.getByText(/Image Search App/i);
  expect(linkElement).toBeInTheDocument();
});

test("should contain searchbox component", () => {
  const container = shallow(<SearchBox />);
  expect(container.html()).toMatchSnapshot();
});

test("should contain gallery component", () => {
  const props = { images: [] };
  const container = shallow(<Gallery {...props} />);
  expect(container.html()).toMatchSnapshot();
});
