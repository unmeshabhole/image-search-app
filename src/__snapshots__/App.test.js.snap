// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should contain gallery component 1`] = `"<div><div class=\\"grid-container\\"></div></div>"`;

exports[`should contain searchbox component 1`] = `"<div class=\\"search-container\\"><input type=\\"text\\" placeholder=\\"Search for image\\" value=\\"\\"/></div>"`;
